gh-105578: Deprecate `AnyStr` in the docs (#107116)
python/cpython: Doc/whatsnew/3.13.rst

    has yet to be supported by any major type checkers.
    (Contributed by Alex Waygood in :gh:`106309`.)

  * :class:`typing.AnyStr` is deprecated in docs since Python 3.13. The intention
    is to remove ``AnyStr`` from ``__all__`` and emit a :exc:`DeprecationWarning`

>     in Python 3.16, and to remove ``AnyStr`` entirely in Python 3.18.

Reviewer: ```suggestion
  * :data:`typing.AnyStr` is deprecated. In Python 3.16,
    it will be removed from ``typing.__all__``, and a :exc:`DeprecationWarning`
    will be emitted when it is imported or accessed.
    It will be removed entirely in Python 3.18.
```

>     (Contributed by Michael The in :gh:`107116`.)

Reviewer: ```suggestion
    Use the new :ref:`type parameter syntax <type-params>` instead.
    (Contributed by Michael The in :gh:`107116`.)
```

* :class:`array.array` ``'u'`` type (:c:type:`wchar_t`):
  use the ``'w'`` type instead (``Py_UCS4``).

Pending Removal in Python 3.18
------------------------------

> * :class:`typing.AnyStr`: Use the new type parameter syntax instead. See :pep:`695`. (:gh:`105578`)

Reviewer: ```suggestion
* :class:`typing.AnyStr`: Use the new
  :ref:`type parameter syntax <type-params>` instead. See :pep:`695`.
  (:gh:`105578`)```
